src/components/MenuQR.jsx

```diff
@@ -0,0 +1,99 @@
+import axios from 'axios';
+import { useEffect, useState, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { View, Text, Platform, ScrollView } from 'react-native';
+import { captureRef } from 'react-native-view-shot';
+import { useParams, Link } from '../Router';
+import { Button } from './ui/Buttons';
+import apiUrl from '../utils/apiUrl';
+import QRCode from 'react-qr-code';
+import createStyles from '../styles/styles';
+
+async function getPageURL(){
+    try {
+        const response = await axios.get(
+            `${apiUrl}/webpageURL/`
+        );
+        const webpageURL = response.data;
+        return webpageURL;
+    } catch (err) {
+        console.error(err);
+        return null;
+    }
+}
+
+const MenuQR = () => {
+    const {t} = useTranslation();
+    const { restaurantId } = useParams();
+    const [menuQRCode, setmenuQRCode] = useState('');
+    const qrViewReference = useRef(null);
+    const [imageUri, setImageUri] = useState('');
+    const styles = createStyles();
+
+    useEffect(() => {
+        const fetchWebpageURL = async () => {
+            const webpageURL = await getPageURL();
+            if (webpageURL) {
+                const urlMenu = webpageURL+'/restaurant/'+restaurantId;
+                setmenuQRCode(urlMenu);
+            } else {
+                console.log('couldnt get webpage URL');
+            }
+        };
+
+        fetchWebpageURL();
+    }, [restaurantId]);
+
+    useEffect(() => {
+        const formImage = async () => {
+            if (qrViewReference.current){
+                const uri = await captureRef(qrViewReference, {
```

Comment:
screenshotting could be avoided with [this trick](https://github.com/rosskhanas/react-qr-code/blob/master/demo/src/Demo.js#L76)

Reply:
Nevermind, this is not actually possible in React Native 😕